Assets/Scripts/Enemy/SpawnPoint.cs

```diff
@@ -1,33 +1,59 @@
+using System.Collections.Generic;
+using System.Linq;
 using MapTools;
 using UnityEngine;
+using Random = UnityEngine.Random;
 
 namespace Enemy
 {
     public class SpawnPoint : MonoBehaviour
     {
-        [Header("Spawn Interval in Seconds")] public float spawnInterval;
-        [SerializeField] private GameObject[] enemies;
+        [SerializeField] private GameObject[] trash;
+        [SerializeField] private GameObject[] bosses;
+        public int WaveCount { private set; get; }
 
+        private bool _spawnBoss;
+        private float _spawnInterval;
         private float _spawnTimer;
 
+        public void InitPoint( bool spawnBoss, float spawnInterval, int waveCount)
+        {
+            _spawnBoss = spawnBoss;
+            _spawnInterval = spawnInterval;
+            WaveCount = waveCount;
+            _spawnTimer = _spawnInterval;
+        }
+
         private void Update()
         {
-            if (spawnInterval == 0)
-            {
-                return;
-            }
+            if (_spawnInterval == 0) return;
+            if (WaveCount == 0) return;
 
             _spawnTimer -= Time.deltaTime;
             if (!(_spawnTimer <= 0)) return;
-            SpawnEnemy();
-            _spawnTimer = spawnInterval;
+
+            if (WaveCount == 1 && _spawnBoss)
+            {
+                SpawnEnemy(bosses, 1);
```

Comment:
the number of bosses should not be hardcoded to 1

Reply:
done